src/coreclr/src/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.cs

```diff
@@ -235,8 +287,13 @@ public void RegisterAsGlobalInstance()
             {
                 throw new InvalidOperationException(SR.InvalidOperation_ResetGlobalComWrappersInstance);
             }
+
+            SetGlobalInstanceRegistered();
         }
 
+        [DllImport(RuntimeHelpers.QCall)]
```

Comment:
I think this function is also a prime candidate for the [`SuppressGCTransition`](https://github.com/dotnet/runtime/issues/30741) attribute.